src/components/structures/RoomView.js

```diff
@@ -54,11 +54,15 @@ module.exports = React.createClass({
     propTypes: {
         ConferenceHandler: React.PropTypes.any,
 
-        roomId: React.PropTypes.string.isRequired,
-
-        // if we are referring to this room by a given alias (e.g. in the URL), track it.
-        // useful for joining rooms by alias correctly (and fixing https://github.com/vector-im/vector-web/issues/819)
-        roomAlias: React.PropTypes.string,
+        // the ID for this room (or, if we don't know it, an alias for it)
+        //
+        // XXX: if this is an alias, we will display a 'join' dialogue,
+        // regardless of whether we are already a member, or if the room is
+        // peekable. Currently there is a big mess, where at least four
+        // different components (RoomView, MatrixChat, RoomDirectory,
+        // SlashCommands) have logic for turning aliases into rooms, and each
+        // of them do it differently and have different edge cases.
```

Comment:
Presumably this is safe because we never actually instantiate RoomView with an alias yet? That said I'd probably rather avoid merging things in where the interface is there but not functional yet as it could trip people up easily. Would it make more sense to make RoomView support both aliases and IDs and then eliminate the duplicate alias lookups once that's in place?